src/app/core/components/manage-tags/manage-tags.component.ts

```diff
@@ -14,6 +14,7 @@ export class ManageTagsComponent implements OnInit {
 
   public editingTagIds: number[] = [];
   public filter: string = '';
+  public originalTagNames: string[] = [];
```

Comment:
I think we want a [map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) instead of an array here. For example, editing a tag with a `tagId` of 1234 would, I think, resize this array to size 1234 (or really, 1235, because zero-indexed)... and the later `delete` will not affect the size.

Example from repl:
```
$ node
Welcome to Node.js v14.18.3.
Type ".help" for more information.
> x = [1, 2, 3]
[ 1, 2, 3 ]
> x[5] = 5
5
> x[5]
5
> x
[ 1, 2, 3, <2 empty items>, 5 ]
> x.length
6
> delete x[5]
true
> x
[ 1, 2, 3, <3 empty items> ]
> x.length
6
```